uploaders/fileuploader.go

```diff
@@ -2,20 +2,54 @@ package uploaders
 
 import (
 	"fmt"
+	"io"
+	"os"
+	"path/filepath"
 
+	"github.com/bitrise-io/go-utils/log"
+	"github.com/bitrise-io/go-utils/pathutil"
 	"github.com/bitrise-steplib/steps-deploy-to-bitrise-io/deployment"
 )
 
+const snapshotFileSizeLimitInBytes = 1024 * 1024 * 1024
+
 // DeployFile ...
 func DeployFile(item deployment.DeployableItem, buildURL, token string) (ArtifactURLs, error) {
+	pth := item.Path
 	fileSize, err := fileSizeInBytes(item.Path)
 	if err != nil {
-		return ArtifactURLs{}, fmt.Errorf("get file size: %s", err)
+		return ArtifactURLs{}, fmt.Errorf("get file size: %w", err)
+	}
+
+	// TODO: This is a workaround to avoid uploading a file that is being modified during the upload process,
+	//  which can cause an issue like: request body larger than specified content length at file upload.
+	deploySnapshot := false
+	if fileSize <= snapshotFileSizeLimitInBytes {
+		snapshotPth, err := createSnapshot(pth)
+		if err != nil {
+			log.Warnf("failed to create snapshot of %s: %s", pth, err)
+		} else {
+			defer func() {
+				if err := os.Remove(snapshotPth); err != nil {
+					log.Warnf("Failed to remove snapshot file: %s", err)
+				}
+			}()
+			pth = snapshotPth
+			deploySnapshot = true
+		}
+	}
+
+	if deploySnapshot {
+		log.Printf("Deploying file snapshot: %s", pth)
```

Comment:
```suggestion
		log.Printf("Deploying snapshot of original file: %s", pth)
```
How about this? I think it provides a bit more context to users who don't know the behind-the-scenes stuff